vsphere/datadog_checks/vsphere/vsphere.py

```diff
@@ -474,6 +474,158 @@ def query_metrics_wrapper(self, query_specs):
         )
         return metrics_values
 
+    def collect_vsan_metrics(self):
+        # type: () -> None
+        self.log.debug("Starting vsan metrics collection (query start time: %s).", self.latest_event_query)
+        latest_metric_time = None
+        collect_start_time = get_current_datetime()
+        try:
+            t0 = Timer()
+            new_health_metrics, new_performance_metrics = self.query_vsan_metrics(
+                collect_start_time - dt.timedelta(hours=2)
+            )
+            self.gauge(
+                'vsphere.vsan.cluster.time',
+                t0.total(),
+                tags=self._config.base_tags,
+                raw=True,
+                hostname=self._hostname,
+            )
+
+            for cluster in new_health_metrics:
+                for metric_name, metric_value in cluster.items():
+                    list_of_tags = ["{}:{}".format(key, value) for key, value in metric_value.items()]
+                    self.gauge(
+                        metric_name,
+                        1,
+                        tags=list_of_tags + self._config.base_tags,
+                        raw=True,
+                        hostname=self._hostname,
+                    )
+
+            for cluster in new_performance_metrics:
+                for entity_type in cluster:
+                    if len(entity_type.value) > 0:
+                        resource_type = entity_type.value[0].metricId.dynamicProperty[0][0]
+                        for given_metric in entity_type.value:
+                            self.log.debug(
+                                "Processing metric with type:%s",
```

Comment:
Can we inlcude more information in this log line like metric name?

Reply:
Added more information.